ddtrace/contrib/django/patch.py

```diff
@@ -377,23 +378,70 @@ def traced_get_response(django, pin, func, instance, args, kwargs):
 
             response = None
 
+            def blocked_response():
+                response = HttpResponseForbidden(
+                    appsec_utils._get_blocked_template(request_headers.get("Accept")), content_type="text/plain"
+                )
+                response.content = appsec_utils._get_blocked_template(request_headers.get("Accept"))
+                return response
+
             try:
-                if _context.get_item("http.request.blocked", span=span):
-                    # Early block (from IP or SRB blocking) before func could run
-                    response = HttpResponseForbidden(
-                        appsec_utils._get_blocked_template(request_headers.get("Accept")),
+                if config._appsec_enabled:
```

Comment:
Would it be possible to refactor some of this code so it can be reused in other modules? There seems to be a bit of repetition. 

Suggestion (probably terrible but its an idea): Maybe we can add a generator that does ip blocking first, yields, then request blocking, yields, then response blocking. That way we can use the same core logic in other integrations.

Reply:
Probably we could refactor a lot, but we try to keep things simple for now and changes easy to follow. Maybe in a later PR, we'll refactor some parts.